annotator-core/src/main/java/edu/ucr/cs/riple/core/metadata/index/Error.java

```diff
@@ -166,4 +187,19 @@ public int hashCode() {
   public String toString() {
     return "Type='" + messageType + '\'' + ", message='" + message + '\'';
   }
+
+  /**
+   * Collects all resolving fixes in given collection of errors as an immutable set.
+   *
+   * @param errors Collection of errors.
+   * @return Immutable set of fixes which can resolve all given errors.
+   */
+  public static ImmutableSet<Fix> getResolvingFixesOfErrors(Collection<Error> errors) {
+    Map<Fix, Set<Set<String>>> m =
+        errors.stream()
+            .flatMap(error -> error.resolvingFixes.stream())
+            .collect(groupingBy(identity(), mapping(fix -> fix.reasons, Collectors.toSet())));
+    m.forEach((fix, sets) -> sets.forEach(fix.reasons::addAll));
+    return ImmutableSet.copyOf(m.keySet());
```

Comment:
FWIW, I find this stream-based code to be somewhat unreadable.  I guess line 202 is doing some kind of mutation on the fixes?  Probably this one would be clearer with loops and iterators.

Reply:
cae2f74cea9219a4916a54f8fe2c094a4c619fd7

Thank you for the suggestion. It was mutating but wasn't making any problems, however, I rewrote to be more cautious.